src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncHelpers.CoreCLR.cs

```diff
@@ -329,35 +343,55 @@ public static void PostToSyncContext(RuntimeAsyncTask task, SynchronizationConte
 
         private static class RuntimeAsyncTaskCore
         {
+            private unsafe ref struct DispatcherInfo
```

Comment:
Could we get this struct with explicit layout or at least match layout how fields will be ordered by loader? I currently use a native struct during stackwalk to get away from field lookups at that part of the logic, it worked before since we didn't have an objectref in the struct, but now loader will switch order of fields, putting NextContinuation first.